siteimprove/admin/js/siteimprove.js

```diff
@@ -133,20 +165,43 @@
       return result;
     };
 
+  function getDocumentHeight() {
+    const body = document.body,
+    html = document.documentElement;
+
+    const height = Math.max( body.scrollHeight, body.offsetHeight, 
+    html.clientHeight, html.scrollHeight, html.offsetHeight );
+
+    return height;
+  }
+
+  function getCorrectDocument (si_prepublish_data) {
+    const newDiv = document.createElement("div"); 
+    newDiv.setAttribute("id","div_iframe"); 
+    document.body.appendChild(newDiv);
+    newDiv.innerHTML = "<iframe id='domIframe' src="+ si_prepublish_data.url.concat("&si_preview=1") +" style='height:"+getDocumentHeight()+"px;width:100%'></iframe>";
+    const domIframe = document.getElementById("domIframe");
+    domIframe.addEventListener( "load" , () => {
+    const newDocument = domIframe.contentWindow.document;
+    siteimprove.contentcheck_flatdom(
+        newDocument,
+        si_prepublish_data.url,
+        si_prepublish_data.token,
+        function () {
+          $(".si-overlay").remove();
+          document.body.removeChild(newDiv);
+        }
+      );
+    });
+  }
+
     $(".siteimprove-trigger-contentcheck")
       .find("a")
       .on("click", function (evt) {
         var si_prepublish_data = siGetCurrentUrlAndToken();
+        getCorrectDocument(si_prepublish_data);
```

Comment:
Can we make a better name for this than `getCorrectDocument` to be more like `getCleanDocument` and then add a comment above that says "opens an alternative version of this page without wp injected content such as the wp-admin bar and smallbox plugin itself as this is for the DOM we send to Siteimprove"

Reply:
Done.